### ISSUE TITLE
Add ``pnpm`` to package manager choice

### ISSUE THREAD

**Issue by Aperrix** [nuxt/create-nuxt-app#367]
Hi everyone !

Can you please add https://pnpm.js.org/en/ to the package manager choice ?
NPM and Yarn are great but PNPM too 😄

**Comment by clarkdo** [comment]
I think current Nuxt.js doesn't support full functionalities of pnpm.
@pi0 Any idea about pnpm support ?

**Comment by Aperrix** [comment]
> I think current Nuxt.js doesn't support full functionalities of pnpm.
> @pi0 Any idea about pnpm support ?

Yeah Nuxt doesn't support PNPM actually because it's necessary to use [--shamefully-hoist flag](https://pnpm.js.org/en/npmrc#shamefully-hoist) so that all dependencies of Nuxt are installed correctly

**Comment by pi0** [comment]
I agree. We plan to fix that hoist flag problem because we decided to stick with vue/webpack deps. But please wait until we change the core changes. So add option when we really support PNPM (again)

**Comment by stefano-b** [comment]
Hi, any news about this?

`[vue-cli](https://github.com/vuejs/vue-cli/issues/3435)` currently supports `pnpm` by using an `[.npmrc](https://docs.npmjs.com/files/npmrc)` file in the project root with the following content

``` .npmrc
shamefully-hoist=true
```

And it works fine, could this be a possible way to use `pnpm` with Nuxt as well?

**Comment by bodrych** [comment]
@pi0 are there any news? :slightly_smiling_face:

**Comment by pi0** [comment]
Hi @bodrych. I think we should finally add this option with 2.15 release. Can you please help trying via `nuxt-edge` if there are any issues with pnp?

**Comment by bodrych** [comment]
@pi0
There was one warning during installation:
```zsh
WARN Failed to find "/fsevents/2.3.2" in lockfile during hoisting. Next aliases will not be hoisted: fsevents
```
And then:
```
Cannot find module 'vue'
Cannot find module 'ufo'
Cannot find module 'node-fetch'
Cannot find module 'vue-meta'
Cannot find module 'vue-client-only'
Cannot find module 'vue-no-ssr'
Cannot find module 'vue-router'
```
![Screenshot_2021-02-08 Screenshot](https://user-images.githubusercontent.com/1725999/107159914-ce3bfe00-69b4-11eb-841e-6cb2ed889239.png)
I've installed the libraries and now `nuxt-edge` works well :slightly_smiling_face:

**Comment by pi0** [comment]
Thanks for checking @bodrych. For reference, yarn2 has a compat plugin for webpack ([pnp-webpack-plugin](https://github.com/arcanis/pnp-webpack-plugin)) that we used (and makes it working out of the box with 2.15). I'm wondering if there is one for pnpm too (maybe we can make same plugin compatible since logic is same it only gets enabled when `process.versions.pnp` is detected). If we do not ensure pnpm works out of the box and add option, it will confuse users when indirect dependencies like `vue` need to be manually installed

**Comment by bodrych** [comment]
@pi0
Can we put `node-linker=pnp` and `symlink=false` to `.npmrc`? (https://pnpm.js.org/en/npmrc#node-linker).
I do not quite understand the problem (because it works well now), but seems like we also have to inject `process.versions.pnp` variable somewhere in `pnpm` (like `yarn` [do](https://github.com/arcanis/yarn/blob/master/src/util/generate-pnp-map-api.tpl.js#L734)) to make `pnp-webpack-plugin` work, haven't we?

**Comment by pi0** [comment]
> because it works well now

Yes but as you mentioned, it works only when adding vue-app dependencies to `package.json` right? It is only my concern.

> Can we put node-linker=pnp and symlink=false to .npmrc? (https://pnpm.js.org/en/npmrc#node-linker).

Sure we may do this when pnpm option is selected in `create-nuxt-app` .

**Comment by bodrych** [comment]
> it works only when adding vue-app dependencies to `package.json`

There's only `nuxt-edge` in dependencies.

**Comment by pi0** [comment]
Then feel free making a PR @bodrych (we shall wait for 2.15 release to merge. also would be nice if can helping to do more tests. most importantly with indirect deps like when adding `@nuxt/axios` module)

**Comment by essarn** [comment]
Did this get merged with the `2.15` release?

**Comment by loclv** [comment]
It would be nice if you set [Only allow pnpm](https://pnpm.io/only-allow-pnpm) config, to prevent devs from using other package managers.

>whenever someone runs npm install or yarn, they'll get an error instead and installation will not proceed.
